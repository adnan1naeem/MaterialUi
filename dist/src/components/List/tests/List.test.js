"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const _1 = require("../");
const react_test_renderer_1 = __importDefault(require("react-test-renderer"));
const react_1 = __importDefault(require("react"));
test('List component with style props', () => {
    const list = react_test_renderer_1.default.create(react_1.default.createElement(_1.List, { style: { width: 100 } })).toJSON();
    expect(list.props.style).toEqual({ width: 100 });
    expect(list).toMatchSnapshot();
});
test('List component with children props', () => {
    const list = react_test_renderer_1.default.create(react_1.default.createElement(_1.List, null,
        react_1.default.createElement("div", null, "Hello"))).toJSON();
    expect(list.props.children).toEqual(undefined);
    expect(list).toMatchSnapshot();
});
test('ListSuheader with children props', () => {
    const listheader = react_test_renderer_1.default.create(react_1.default.createElement(_1.ListSubheader, null, "I am Header of list")).toJSON();
    expect(listheader.props.children).toEqual(undefined);
    expect(listheader).toMatchSnapshot();
});
test('ListItem  with children props', () => {
    const listitem = react_test_renderer_1.default.create(react_1.default.createElement(_1.ListItem, null, "Hello")).toJSON();
    expect(listitem.props.children).toEqual(undefined);
    expect(listitem).toMatchSnapshot();
});
test('ListItem with   style props', () => {
    const listitem = react_test_renderer_1.default.create(react_1.default.createElement(_1.ListItem, { style: { width: 100 } })).toJSON();
    expect(listitem.props.style).toEqual({ width: 100 });
    expect(listitem).toMatchSnapshot();
});
test('ListItem with primary props', () => {
    const listitem = react_test_renderer_1.default.create(react_1.default.createElement(_1.ListItem, { primary: "hello" }, "Hello")).toJSON();
    expect(listitem.props.primary).toEqual('hello');
    expect(listitem).toMatchSnapshot();
});
test('ListItem with secondary props', () => {
    const listitem = react_test_renderer_1.default.create(react_1.default.createElement(_1.ListItem, { secondary: "Hi" }, "Hello")).toJSON();
    expect(listitem.props.secondary).toEqual('Hi');
    expect(listitem).toMatchSnapshot();
});
//# sourceMappingURL=List.test.js.map