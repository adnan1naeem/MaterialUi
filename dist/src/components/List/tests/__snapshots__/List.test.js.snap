// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component with children props 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
  style={undefined}
>
  <div>
    Hello
  </div>
</ul>
`;

exports[`List component with style props 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
  style={
    Object {
      "width": 100,
    }
  }
/>
`;

exports[`ListItem  with children props 1`] = `
<li
<<<<<<< HEAD
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-18"
=======
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-19"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
  disabled={false}
  onClick={undefined}
>
  <div
<<<<<<< HEAD
    className="MuiListItemText-root-22"
  >
    <span
      className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
=======
    className="MuiListItemText-root-23"
  >
    <span
      className="MuiTypography-root-29 MuiTypography-subheading-36 MuiListItemText-primary-26"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
    >
      Hello
    </span>
  </div>
</li>
`;

exports[`ListItem with   style props 1`] = `
<li
<<<<<<< HEAD
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-18"
=======
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-19"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
  disabled={false}
  onClick={undefined}
  style={
    Object {
      "width": 100,
    }
  }
/>
`;

exports[`ListItem with primary props 1`] = `
<li
<<<<<<< HEAD
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-18"
=======
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-19"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
  disabled={false}
  onClick={undefined}
  primary="hello"
>
  <div
<<<<<<< HEAD
    className="MuiListItemText-root-22"
  >
    <span
      className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
=======
    className="MuiListItemText-root-23"
  >
    <span
      className="MuiTypography-root-29 MuiTypography-subheading-36 MuiListItemText-primary-26"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
    >
      hello
    </span>
    <p
<<<<<<< HEAD
      className="MuiTypography-root-28 MuiTypography-body1-37 MuiTypography-colorTextSecondary-61 MuiListItemText-secondary-26"
=======
      className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-colorTextSecondary-62 MuiListItemText-secondary-27"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
    >
      
    </p>
  </div>
  <div
<<<<<<< HEAD
    className="MuiListItemText-root-22"
  >
    <span
      className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
=======
    className="MuiListItemText-root-23"
  >
    <span
      className="MuiTypography-root-29 MuiTypography-subheading-36 MuiListItemText-primary-26"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
    >
      Hello
    </span>
  </div>
</li>
`;

exports[`ListItem with secondary props 1`] = `
<li
<<<<<<< HEAD
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-18"
=======
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-19"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
  disabled={false}
  onClick={undefined}
  secondary="Hi"
>
  <div
<<<<<<< HEAD
    className="MuiListItemText-root-22"
  >
    <span
      className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
=======
    className="MuiListItemText-root-23"
  >
    <span
      className="MuiTypography-root-29 MuiTypography-subheading-36 MuiListItemText-primary-26"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
    >
      
    </span>
    <p
<<<<<<< HEAD
      className="MuiTypography-root-28 MuiTypography-body1-37 MuiTypography-colorTextSecondary-61 MuiListItemText-secondary-26"
=======
      className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-colorTextSecondary-62 MuiListItemText-secondary-27"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
    >
      Hi
    </p>
  </div>
  <div
<<<<<<< HEAD
    className="MuiListItemText-root-22"
  >
    <span
      className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
=======
    className="MuiListItemText-root-23"
  >
    <span
      className="MuiTypography-root-29 MuiTypography-subheading-36 MuiListItemText-primary-26"
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
    >
      Hello
    </span>
  </div>
</li>
`;

exports[`ListSuheader with children props 1`] = `
<li
  className="MuiListSubheader-root-5 MuiListSubheader-sticky-10 MuiListSubheader-gutters-8"
>
  I am Header of list
</li>
`;
